src/Cli/PluginCommands.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace UpdateTool\Cli;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Robo\Contract\ConfigAwareInterface;
+use Robo\Common\ConfigAwareTrait;
+use UpdateTool\Git\WorkingCopy;
+use UpdateTool\Git\Remote;
+
+/**
+ * Commands used to create pull requests to update the available command line tool version on the platform
+ */
+class PluginCommands extends \Robo\Tasks implements ConfigAwareInterface, LoggerAwareInterface
+{
+    use ConfigAwareTrait;
+    use LoggerAwareTrait;
+    use ApiTrait;
+
+    /**
+     * @command plugin:update
+     */
+    public function pluginUpdate($remote, $options = ['as' => 'default', 'push' => true, 'check' => false])
+    {
+        $api = $this->api($options['as']);
+
+        $url = $this->getConfig()->get("plugins.$remote.repo");
+        if (empty($url)) {
+            throw new \Exception('Plugin cannot be updated; it is missing a source.');
+        }
+
+        $work_dir = $this->getConfig()->get("plugins.$remote.path");
+        $main_branch = $this->getConfig()->get("plugins.$remote.main-branch");
+
+        $plugin_working_copy = WorkingCopy::clone($url, $work_dir, $api);
+        $plugin_working_copy
+            ->addFork('')
+            ->setLogger($this->logger);
+        $this->logger->notice("Check out {project} to {path}.", ['project' => $plugin_working_copy->projectWithOrg(), 'path' => $work_dir]);
+        $plugin_working_copy
+        ->switchBranch($main_branch)
+        ->pull('origin', $main_branch);
+
+        // Get plugin file to parse.
+        $file_to_check = $this->getConfig()->get("plugins.$remote.file-to-check");
+        $query_string = $this->getConfig()->get("plugins.$remote.query-string");
+        $versions_file_path = "$work_dir/$file_to_check";
+        $version_file_contents = file_get_contents($versions_file_path);
+
+        // Check for updates.
+        $api_url = $this->getConfig()->get("plugins.$remote.version-api.url", '') ;
+        $latest_versions = $this->findLatestVersion($api_url);
+
+        // Get current version, compare to updated version, and update file if needed.
+        $version_file_contents = $this->getCurrentVersionUpdateFile($version_file_contents, $query_string, $latest_versions);
+
+        $message = $this->message($remote);
+        // Determine if there are any PRs already open that we should
+        // close. If its contents are the same, then we should abort rather than create the same PR again.
+        // If the contents are different, then we'll make a new PR and close this one.
+        $prs = $api->matchingPRs($plugin_working_copy->projectWithOrg(), $message);
+        if (in_array($message, $prs->titles())) {
```

Comment:
Would it make sense to move this before findLatestVersion to make it faster when an update is not needed?

Also, it looks like the PR title will always be the same, is that right? If so, there's no way to differentiate between exactly the same PR vs PRs with older versions unless I'm missing something

Reply:
Correct, right now it does not have the version in the PR title. I'll make an update to fix that, but that will mean we'll need to keep `findLatestVersion` before this to compare the titles.

Reply:
Fixed this in [2f969d5](https://github.com/pantheon-systems/update-tool/pull/64/commits/2f969d5331c7ae41b7b3cac80464554d5a3bc63e). Tested on [my fork](https://github.com/jspellman814/wp_launch_check/pulls), PRs 16-18